### ISSUE TITLE
Incorrect main export for preact listed when run without conditions

### ISSUE THREAD

**Issue by bgotink** [ljharb/list-exports#77]
List-exports incorrectly adds `preact` as possible entry point when requesting all exports with conditions enabled:

https://github.com/ljharb/list-exports/blob/84a90dcc33bf85b32ef37732e2017ffd1a2146ff/packages/tests/fixtures/preact/expected-without-conditions.json#L9

However, using node 13 <13.7 this fails:

```
$ node
Welcome to Node.js v13.0.1.
Type ".help" for more information.
> require('preact')
Thrown:
Error: Package exports for '/private/var/folders/_d/ch2kc4h960d10cy_2c41qqzw0000gn/T/tmp.4w99c5aIqE/node_modules/preact' do not define a valid '.' target
    at resolveExportsTarget (internal/modules/cjs/loader.js:460:13)
    at resolveExports (internal/modules/cjs/loader.js:393:16)
    at Function.Module._findPath (internal/modules/cjs/loader.js:492:20)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:787:27)
    at Function.Module._load (internal/modules/cjs/loader.js:693:27)
    at Module.require (internal/modules/cjs/loader.js:864:19)
    at require (internal/modules/cjs/helpers.js:74:18) {
  code: 'MODULE_NOT_FOUND'
}
```

That's because the package.json contains the following two export definitions:

https://github.com/ljharb/list-exports/blob/84a90dcc33bf85b32ef37732e2017ffd1a2146ff/packages/tests/fixtures/preact/project/package.json#L128-L134

https://github.com/ljharb/list-exports/blob/84a90dcc33bf85b32ef37732e2017ffd1a2146ff/packages/tests/fixtures/preact/project/package.json#L169

Node only matches the first, which it doesn't understand, so it throws an error.

list-exports ignores the `.` export because it contains condtions, but it adds the `.` export anyway when it encounters the `./` later on.

**Comment by ljharb** [comment]
Hmm, indeed you’re right (on node v13.2-13.6, this fails because it only looks for “default”; on v13.0-v13.1, it fails because node only looks for the string form).

**Comment by ljharb** [comment]
I think the slash logic needs to avoid adding paths that have previously been specified in the LHS.

**Comment by ljharb** [comment]
Fixed in #78 / fixed in #81, i believe.
